### ISSUE TITLE
Add more "package mangement" features

### ISSUE THREAD

**Issue by bfiete** [beefytech/Beef#409]
The eventual direction is to have more Cargo-like management features in BeefIDE/BeefBuild.

**Comment by RogueMacro** [comment]
What do you think about a package registry and do you have something in mind for naming? Do you see this coming to Beef sometime in the future?

**Comment by bfiete** [comment]
Yes, this is probably going to be the next "big feature" after metaprogramming. Yes, a registry. For naming, I don't plan on any special name, it's just another set of features in the language/tools.

**Comment by RogueMacro** [comment]
I don't know if you planned to do this, but maybe when you add packages to your workspace they could be under a separate folder along with the workspace. Then many packages don't clutter up your Workspace explorer.

**Comment by bfiete** [comment]
It seems like you're thinking a "package" is different than a "project" in some way?

**Comment by RogueMacro** [comment]
I was thinking that a project would be a project you created yourself. Like a directory can have multiple projects, but the external/BeefLibs packages/projects you add would be packages. I don't see a problem right now, but if theoretically, you had many projects or projects that depend on projects, it would be a nice feature. Or maybe just to have a cleaner workspace.

**Comment by xposure** [comment]
It would depend how far the package management feature goes, if its something like nuget, you wouldn't want any of the package stuff in the git repo and it would be managed by some sort of xml file. Then you would have to do a package restore to get all the dependencies (which the IDE could just do as a pre build event). 

Keeping it under a `package` folder allows you to easily add it to the git ignore file (or whatever source control provider is being used). They may be treated as another project, but for source control management, they aren't really another project.
